homeassistant/components/light/__init__.py

```diff
@@ -369,3 +370,16 @@ def state_attributes(self):
     def supported_features(self):
         """Flag supported features."""
         return 0
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Component added, restore_state using platforms."""
+        state = get_last_state(self)
+        if state:
```

Comment:
Your view on moving this to `homeassistant.helpers.restore_state`?

Two reasons:
* `recorder` does not explicitly have to request the cache, but `get_last_state` handles this.
* Many components & platforms (possibly every sensor) will import this "helper"

The cache function can then be a _private